File: perl/sorting/selection_sort/selection_sort.pl
#!/usr/bin/env perl

my @int_arr = (4, 6, 3, 2, 6, 8, 1, 5, 9, 5, 2, 3, 8, 1, 2, 5, 0, 9);

my @char_arr = ('a', 'n', 'e', 't', 'u', 'h', 'e', 's', 'x', 'h', 'i', 'o', 'j', 'g', 'f', 'd', 'e', 'g', 'z', 'a',
               'e', 'w', 'f', 'h', 'k', 'h', 'f', 'l', 'a', 'q', 'w', 'r', 'y', 'h', 'f');


sub sort_arr {
    my @arr = @_;
    my $len = scalar(@arr);
    my $min;

    # Loop through unsorted array
    for (my $i = 0; $i < $len; $i++) {
    for (my $i = 0; $i < $len; $i++) {
	# Find min in unsorted array
	for (my $j = $i + 1; $j < $len; $j++) {
	    if ($arr[$j] < $arr[$i]) {
        	# Swap min element with first element
		$min = $arr[$j];
		$arr[$j] = $arr[$i];
		$arr[$i] = $min;
	    }
	}

	break;
    }

    print_arr(@arr);
}

sub sort_char {
    my @arr = @_;
    my $len = scalar(@arr);
    my $min;

    # Loop through unsorted array
    for (my $i = 0; $i < $len; $i++) {
	# Find min in unsorted array
	for (my $j = $i + 1; $j < $len; $j++) {
	    my $comp = $arr[$j] cmp $arr[$i];
	    if ($comp == -1) {
        	# Swap min element with first element
		$min = $arr[$j];
		$arr[$j] = $arr[$i];
		$arr[$i] = $min;
	    }
	}

	break;
    }

    print_arr(@arr);
}



sub print_arr {
    my @arr = @_;

    foreach my $item (@arr) {
	print $item . ", ";
    }

    print("\n");
}


print_arr(@int_arr);
sort_arr(@int_arr);

print("\n");

print_arr(@char_arr);
sort_char(@char_arr);

File: python/sorting/selection_sort/selectionSort.py
#!/usr/bin/python3

def sort_array(arr):
    # Loop through unsorted array
    for i in range(len(arr)):
        # Find min in unsorted array
        min_index = i;
        for j in range(i + 1, len(arr)):
            if arr[min_index] > arr[j]:
                min_index = j

        # Swap min element with first element
        arr[i], arr[min_index] = arr[min_index], arr[i]

    print_sorted_array(arr)


def print_sorted_array(sorted_arr):
    print('Sorted Array:')
    print(sorted_arr),


unsortedNum = [4, 6, 3, 2, 6, 8, 1, 5, 9, 5, 2, 3, 8, 1, 2, 5, 0, 9]
unsortedAlpha = ['a', 'n', 'e', 't', 'u', 'h', 'e', 's', 'x', 'h', 'i', 'o', 'j', 'g', 'f', 'd', 'e', 'g', 'z', 'a',
                 'e', 'w', 'f', 'h', 'k', 'h', 'f', 'l', 'a', 'q', 'w', 'r', 'y', 'h',
                 'f']

print('Unsorted Integer Array:')
print(unsortedNum)
sort_array(unsortedNum)

print('------------------------------------')
print('Unsorted Alpha Array:')
print(unsortedAlpha)
sort_array(unsortedAlpha)
